File: java/src/main/java/io/fivexer/sdk/Team.java
package io.fivexer.sdk;

import io.fivexer.sdk.internal.Json;
import io.fivexer.sdk.model.TeamPresence;

/**
 * The supervisor view of presence: who is working, on break, or paused. Paused is an operator
 * action; on-break is the worker's own, and the two are reported separately. Requires the
 * control plane.
 */
public final class Team {

    private final Fivexer client;

    Team(Fivexer client) {
        this.client = client;
    }

    public TeamPresence presence() {
        return presence(null);
    }

    /** @param teamId narrow the view to one crew, or null for the whole workspace */
    public TeamPresence presence(String teamId) {
        return client.request("GET", "/team/presence", null, Json.query("teamId", teamId),
                TeamPresence.class, false);
    }
}

File: php/src/Resource/Team.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Resource;

use Fivexer\SDK\Fivexer;
use Fivexer\SDK\Model\TeamPresence;

/**
 * The supervisor view of presence: who is working, on break, or paused. Paused is an operator
 * action; on-break is the worker's own, and the two are reported separately. Requires the
 * control plane.
 */
final class Team
{
    public function __construct(private readonly Fivexer $client)
    {
    }

    /** @param string|null $teamId narrow the view to one crew, or null for the whole workspace */
    public function presence(?string $teamId = null): TeamPresence
    {
        return TeamPresence::fromArray(
            $this->client->request('GET', '/team/presence', null, ['teamId' => $teamId]) ?? []
        );
    }
}
